File: recursive/recurse.java

/*Recursion in Java*/

public class Main {
  public static void main(String[] args) {
    for(int i=0;i<1000000;i++)
    System.out.println(sum(100));
  }

  public static int sum(int k) {
    if (k > 0) {
      return k + sum(k - 1);
    } else {
      return 0;
    }
  }
}

File: recursive/recurse.c

//Recursion in the C language

#include <stdio.h>

int sum(int k) {
  if (k > 0) {
    return k + sum(k - 1);
  } else {
    return 0;
  }
}

int main() {

  for (int i = 0; i < 1000000; i++)
  {
      int result = sum(100);
      printf("%d\n",result);
  }
  return 0;
}
